File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;
class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        if (cacheSize == 0) {
            return cities.length * 5;
        }
        ArrayList<String> list = new ArrayList<>();
        for(int i = 0; i < cities.length; i++) {
            cities[i] = cities[i].toLowerCase();
            if(list.contains(cities[i])) {
                list.remove(cities[i]);
                list.add(cities[i]);
                answer++;
                continue;
            }
            if (list.size() < cacheSize) {
                list.add(cities[i]);
                
            }
            else {
                list.remove(0);
                list.add(cities[i]);
                
            }
            answer += 5;
        }
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    var answer = 0;
    const cache = [];
    if (cacheSize === 0) return cities.length * 5;
    for (let i = 0; i < cities.length; i++) {
        cities[i] = cities[i].toLowerCase();
        const city = cities[i];
        if (!cache.includes(city)) {
            answer += 5;
            if (cache.length === cacheSize) {
                cache.shift();
                cache.push(city);
            } else cache.push(city);
        } else {
            const index = cache.findIndex((v) => v === city);
            const recent = cache.splice(index,1)[0];
            cache.push(recent);
            answer++;
        }
    }
    return answer;
}